internal/controllers/topology/cluster/structuredmerge/twowayspatchhelper.go

```diff
@@ -233,7 +234,7 @@ func isNil(i interface{}) bool {
 	}
 	switch reflect.TypeOf(i).Kind() {
 	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
-		return reflect.ValueOf(i).IsValid() && reflect.ValueOf(i).IsNil()
+		return util.IsNil(i)
```

Comment:
Should the util func be this entire func instead?

Looks like a safer way to check for nil to me